### PR TITLE
applications: nrf_desktop: Switch to hardware v03

### PR DESCRIPTION
**Pull request by MarekPieta**
Battery charging signal is inversed for hardware v03.

### CODE REVIEW

File: applications/nrf_desktop/src/hw_interface/Kconfig [hunk 1]
```diff
@@ -184,6 +184,14 @@ config DESKTOP_BATTERY_CHARGER_ENABLE_PIN
 	help
 	  A pin which enables battery charging.
 
+config DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED
+	bool "Inverse charging enable signal"
+	depends on DESKTOP_BATTERY_CHARGER_DISCRETE
+	default n
```
Comment (SebastianBoe) [inline applications/nrf_desktop/src/hw_interface/Kconfig]: "default n" is the default so it is best-practice to omit this line.

File: applications/nrf_desktop/src/hw_interface/battery_charger.c [hunk 2]
```diff
@@ -113,16 +113,20 @@ static int cso_pin_control(bool enable)
...
+
 	return gpio_pin_write(gpio_dev,
 			      CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_PIN,
-			      1);
+			      charging);
 }
 
 static int charging_disable(void)
 {
+	u32_t charging = CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED ? 1 : 0;
```
Comment (SebastianBoe) [inline applications/nrf_desktop/src/hw_interface/battery_charger.c]: isn't it possible for CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_INVERSED to be undefined?

Even if it is not, IMHO we should wrap it with IS_ENABLED() to be safe (it converts undefined macros to 0).

File: applications/nrf_desktop/src/hw_interface/battery_charger.c [hunk 3]
```diff
@@ -113,16 +113,20 @@ static int cso_pin_control(bool enable)
...
+
 	return gpio_pin_write(gpio_dev,
 			      CONFIG_DESKTOP_BATTERY_CHARGER_ENABLE_PIN,
-			      0);
+			      charging);
```
Comment (SebastianBoe) [inline applications/nrf_desktop/src/hw_interface/battery_charger.c]: Please have charging_enable and charging_disable call a common function that takes a bool as an argument to reduce the copy-paste here.

File: applications/nrf_desktop/src/hw_interface/Kconfig (hunk 1 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/hw_interface/Kconfig]: done

File: applications/nrf_desktop/src/hw_interface/battery_charger.c (hunk 3 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/hw_interface/battery_charger.c]: ok

File: applications/nrf_desktop/src/hw_interface/battery_charger.c (hunk 2 above)
Comment (MarekPieta) [reply applications/nrf_desktop/src/hw_interface/battery_charger.c]: Ok, I add it

**Comment by MarekPieta** [comment]
I made the changes, we need to merge [PR](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/881) first and rebase

**Comment by MarekPieta** [comment]
@SebastianBoe, could you rereview ?

**Comment by MarekPieta** [comment]
@Qbicz could you review?

**Comment by pdunaj** [comment]
Hi @wiminordic , are we ready to merge?

**Pull request merged**